Sofa/framework/Simulation/Core/src/sofa/simulation/VisualVisitor.cpp

```diff
@@ -49,9 +49,6 @@ Visitor::Result VisualVisitor::processNodeTopDown(simulation::Node* node)
 
 Visitor::Result VisualDrawVisitor::processNodeTopDown(simulation::Node* node)
 {
-    // don't draw if specified not to do so in the user interface
```

Comment:
Can you argue a bit why you want to call the rendering function of a visual model while the showvisualmodel is disabled ?

Reply:
https://github.com/sofa-framework/sofa/pull/4044#issuecomment-1637217987

Reply:
Well to me removing the test is not the way to fix that. VisualDrawVisitor shouldn't be used to call the drawing of non visual object. Doing that has the consequence of maintaining the fuzziness of use case between draw() and drawVisual(). 

So ok to merge  as is to restore the rendering of object using draw() but a more serious refactoring is needed there.

Reply:
Ok so if I understand this is already merged. So let's make this PR to ready. 

EDIT: The merged fixed is not one that should have been done (for two reason 1) because the VisualDrawVisitor is not supposed to process non VisualModel and 2) only the tests processObject(BaseObject) should be removed not the one that process only visual objects). But as it is merge let's move forward, maybe one day someone will refactor this code to split the draw() and drawVisual().